### PR TITLE
build: fix mingw miniupnpc cflags

### PR DESCRIPTION
**Pull request by fanquake**
Pulls in a patch I've upstreamed to miniupnpc so that we properly pass our cflags when building it for mingw. See https://github.com/miniupnp/miniupnp/pull/619. Also set `D_WIN32_WINNT` to `0x0601` to match libevent, configure etc. Previously it was being set to `0X501`.

Guix Build (x86_64 / arm64):
```bash
39a66c473a45b83ca85500b32ccf8f30d4ae80f965ca064566ee9fd84a51964b  guix-build-859644b3c855/output/aarch64-linux-gnu/SHA256SUMS.part
7b0515e422f350cb23f4f0b2f87eaa1b30d1c80389da6f1cbe700794902c88e9  guix-build-859644b3c855/output/aarch64-linux-gnu/bitcoin-859644b3c855-aarch64-linux-gnu-debug.tar.gz
192253fb387a2216b6d63d47a18e34bfa284874488d7ebc6ba656ca76a905519  guix-build-859644b3c855/output/aarch64-linux-gnu/bitcoin-859644b3c855-aarch64-linux-gnu.tar.gz
0a8b5c77928a46e62dff85cf73ca9e932560533b99ef1ec374be00516f9e1183  guix-build-859644b3c855/output/arm-linux-gnueabihf/SHA256SUMS.part
d1b01b36d7092d63ab84877e05a973d915d177dbc618fe00eeaff86295032750  guix-build-859644b3c855/output/arm-linux-gnueabihf/bitcoin-859644b3c855-arm-linux-gnueabihf-debug.tar.gz
50d9cd81a4a37fbd5c22ee8f1b8398a836879bda1b514a9ed3d0bcd6fd3de41f  guix-build-859644b3c855/output/arm-linux-gnueabihf/bitcoin-859644b3c855-arm-linux-gnueabihf.tar.gz
e590d6dc6687c744b4067af330a7fe44110da4972c46f0262c39ce03e2aa6ac5  guix-build-859644b3c855/output/arm64-apple-darwin/SHA256SUMS.part
6b9b97a1f6ead6d2b70d706ba39b11f36c2929962afc0d52404f2341d412d4e7  guix-build-859644b3c855/output/arm64-apple-darwin/bitcoin-859644b3c855-arm64-apple-darwin-unsigned.dmg
de6be985a4e1b11c6450c388b54be4fff3dc3a78e528cb628623ee4a8ea249f4  guix-build-859644b3c855/output/arm64-apple-darwin/bitcoin-859644b3c855-arm64-apple-darwin-unsigned.tar.gz
940024658b9387040ceb26535dbd1ed7edb3709106f6e25d5d3720ed90bbbb2e  guix-build-859644b3c855/output/arm64-apple-darwin/bitcoin-859644b3c855-arm64-apple-darwin.tar.gz
67c8fcdd31dca595e5c6b72e597a135718dd50ce1f062cb18d181520c13d3013  guix-build-859644b3c855/output/dist-archive/bitcoin-859644b3c855.tar.gz
4b1954953913d1387589873a8e9dc9765f0f300c125270046da95a23c43aa069  guix-build-859644b3c855/output/powerpc64-linux-gnu/SHA256SUMS.part
5b12ab7e3a7fc162912e67e026646ee5d4c92ef804525504f188d9a569af7d67  guix-build-859644b3c855/output/powerpc64-linux-gnu/bitcoin-859644b3c855-powerpc64-linux-gnu-debug.tar.gz
fa4debb24dbb4c9c515ad3bfa2e5cb9bd686f1062157bd4480d076ef35d7ea9d  guix-build-859644b3c855/output/powerpc64-linux-gnu/bitcoin-859644b3c855-powerpc64-linux-gnu.tar.gz
bafad2f56aad4edd25e49e4cc658811cf87f32a2a2e013672659d410d7a8cac8  guix-build-859644b3c855/output/powerpc64le-linux-gnu/SHA256SUMS.part
96318cac800dc7ee86229754a76047dececd62ecd962362e6dbd1d694bc8c17a  guix-build-859644b3c855/output/powerpc64le-linux-gnu/bitcoin-859644b3c855-powerpc64le-linux-gnu-debug.tar.gz
461415ee7bf67e2c59bfce3a09736b3b0ecdc7c81751ce2bed19369237450154  guix-build-859644b3c855/output/powerpc64le-linux-gnu/bitcoin-859644b3c855-powerpc64le-linux-gnu.tar.gz
5c9d19f6af5d1fab7de7496bbdcd1cd266abd6c0e2c226b362dab2582c94a33a  guix-build-859644b3c855/output/riscv64-linux-gnu/SHA256SUMS.part
04492f5ce121ba09672119cc8861ee30ecdb814b34726c9c5d7971c528209a55  guix-build-859644b3c855/output/riscv64-linux-gnu/bitcoin-859644b3c855-riscv64-linux-gnu-debug.tar.gz
5fb28111b49e73fc53db2805500c275a2e6321c3e180695eb813675629dcd64c  guix-build-859644b3c855/output/riscv64-linux-gnu/bitcoin-859644b3c855-riscv64-linux-gnu.tar.gz
3f48c1c2ba77d4fda725225f6c9b5ab7f3aae244c8abb354407cc73d6547d983  guix-build-859644b3c855/output/x86_64-apple-darwin/SHA256SUMS.part
6dcfb5a4af350466fb7f9319e02fd4bcef66e015116c9eda8aff03b3ac53d109  guix-build-859644b3c855/output/x86_64-apple-darwin/bitcoin-859644b3c855-x86_64-apple-darwin-unsigned.dmg
87112cc5f2d02c16614f6d8df41af6f3ea3c765eb6d196d68d2b514b6bd317bc  guix-build-859644b3c855/output/x86_64-apple-darwin/bitcoin-859644b3c855-x86_64-apple-darwin-unsigned.tar.gz
b245a4d4881a679f2c91e0c7fd5466d6b93313289e609dbce4e1009b7591332e  guix-build-859644b3c855/output/x86_64-apple-darwin/bitcoin-859644b3c855-x86_64-apple-darwin.tar.gz
a4c3449aa9d8a8e1c8b0532f6e74845c40f90ad4186f480d5bb2750e184efc10  guix-build-859644b3c855/output/x86_64-linux-gnu/SHA256SUMS.part
12a32f25d6ef7f60023dd3053b21e31131350b6afa01aeab25e53d1928b1a0f5  guix-build-859644b3c855/output/x86_64-linux-gnu/bitcoin-859644b3c855-x86_64-linux-gnu-debug.tar.gz
93d603d89c9251e93992e3c0e176b094d981b020ef1d4bbfbff21806e50b962d  guix-build-859644b3c855/output/x86_64-linux-gnu/bitcoin-859644b3c855-x86_64-linux-gnu.tar.gz
79df55f4ca8f972665fd55b43550b2ab9ffafa8fc8de1335b324f4af047c6788  guix-build-859644b3c855/output/x86_64-w64-mingw32/SHA256SUMS.part
8d837e96595ab7337736da334d940ffc9d15215141f176804a528fe9e21f490a  guix-build-859644b3c855/output/x86_64-w64-mingw32/bitcoin-859644b3c855-win64-debug.zip
6c9541524f1d54eceb3265c6e79d62502fdc0c2e5263719a0ca357988d7ed718  guix-build-859644b3c855/output/x86_64-w64-mingw32/bitcoin-859644b3c855-win64-setup-unsigned.exe
7566ab4ee53092e81c3079db955d85c8d574cbde2be21526d45619076ffcd264  guix-build-859644b3c855/output/x86_64-w64-mingw32/bitcoin-859644b3c855-win64-unsigned.tar.gz
32164cfa7c06ead63305485653f37d74c6ada82d28b79f58e66faf6e72e130bb  guix-build-859644b3c855/output/x86_64-w64-mingw32/bitcoin-859644b3c855-win64.zip
```

### CODE REVIEW

**Comment by hebasto** [comment]
Instead of introducing a patch into our repository, why not bumping `miniupnpc` sources as we have done for `libnatpmp` in #21209?

**Comment by fanquake** [comment]
> why not bumping miniupnpc sources

I haven't tested them, and don't currently plan to. The 2.2.3 tag is also broken out of the box.

**Comment by fanquake** [comment]
> why not bumping miniupnpc sources

There's also no release/tag that would include my patch. So are you suggesting we move to using the master branch of miniupnpc? I doubt that is good idea, or would be sources that are particularly well tested. The only reason we do that for libnatpmp is because the author hasn't made a release in 7 years.

**Comment by jarolrod** [comment]
GUIX hashes

x86:
```
a614b6d36480ece6a5171efe965031960a0d54350fd13dc8c50b2fb498047db2  guix-build-1a332c78dbc7/output/aarch64-linux-gnu/SHA256SUMS.part
57b35f41a3e3ee9b87aa844d0fe3db0d3055a7963df3e50641c6f7c6f5d67f34  guix-build-1a332c78dbc7/output/aarch64-linux-gnu/bitcoin-1a332c78dbc7-aarch64-linux-gnu-debug.tar.gz
1eee2a1e99bbf0113994fb4d69b93c645d08eca60ddba4f2defd74d91b5a8990  guix-build-1a332c78dbc7/output/aarch64-linux-gnu/bitcoin-1a332c78dbc7-aarch64-linux-gnu.tar.gz
261d5a0087d5088a6c9901405dd8b8d0da0650709cda47284f18c13cf5ba0086  guix-build-1a332c78dbc7/output/arm-linux-gnueabihf/SHA256SUMS.part
24ec9d18985fdf03b3c91d82e81eee05d41426672a3e3c12a6142994d7cfdb84  guix-build-1a332c78dbc7/output/arm-linux-gnueabihf/bitcoin-1a332c78dbc7-arm-linux-gnueabihf-debug.tar.gz
5c74fb3da500aae8144f59f47772cafdbcb7a7a29c971fdd7173d89c7e639150  guix-build-1a332c78dbc7/output/arm-linux-gnueabihf/bitcoin-1a332c78dbc7-arm-linux-gnueabihf.tar.gz
34cb3ff54d6e59bc7b95f5d93c1a797c2c98204787bdcac7345c193436127b21  guix-build-1a332c78dbc7/output/arm64-apple-darwin/SHA256SUMS.part
4c9af65047c07984008a9e91f74b3f78f39263819471627cd9c6c7a7dd9938ef  guix-build-1a332c78dbc7/output/arm64-apple-darwin/bitcoin-1a332c78dbc7-arm64-apple-darwin-unsigned.dmg
a57c8f2da2235af5082300eea60f3040aa613d8a34e9720ba15708e2a428ab03  guix-build-1a332c78dbc7/output/arm64-apple-darwin/bitcoin-1a332c78dbc7-arm64-apple-darwin-unsigned.tar.gz
5ec93b3dd36ec9a42f3fa13ee152cdf3bbfd7d3a33a26ebfb63ac8391ea452d3  guix-build-1a332c78dbc7/output/arm64-apple-darwin/bitcoin-1a332c78dbc7-arm64-apple-darwin.tar.gz
e7e782aa915eab0f51881dfc9cfbec42277090cbd01c3ff262d24228849d2153  guix-build-1a332c78dbc7/output/dist-archive/bitcoin-1a332c78dbc7.tar.gz
ea2e95e54cde30468a8c400e189fcb19b48dc7217dfa54871ce6b1eda654bf94  guix-build-1a332c78dbc7/output/powerpc64-linux-gnu/SHA256SUMS.part
ee8a23b78e21438d181818e77b3cf96bf1f337dd6585c282cf889ce7c6debc52  guix-build-1a332c78dbc7/output/powerpc64-linux-gnu/bitcoin-1a332c78dbc7-powerpc64-linux-gnu-debug.tar.gz
2ac5cadf0f2a3f9456777c900956e5210850f5415c36024b26d1cc5f179f7bc5  guix-build-1a332c78dbc7/output/powerpc64-linux-gnu/bitcoin-1a332c78dbc7-powerpc64-linux-gnu.tar.gz
763219587b6eb81d4487ad8ad2a2601f05cf3235616809a464d1aae47ac8cc5d  guix-build-1a332c78dbc7/output/powerpc64le-linux-gnu/SHA256SUMS.part
5de7cd11346c7ac6f40e4dc79a7f5fad6333b98d72ae65411a7bb1e4b9e0438a  guix-build-1a332c78dbc7/output/powerpc64le-linux-gnu/bitcoin-1a332c78dbc7-powerpc64le-linux-gnu-debug.tar.gz
8b6807dbbf11fdec03da67974e93e247719148cb214f948d2fcf7a0a29ebc121  guix-build-1a332c78dbc7/output/powerpc64le-linux-gnu/bitcoin-1a332c78dbc7-powerpc64le-linux-gnu.tar.gz
ccbefa5851b91df1effb16ed1d22982f69b43bff2582bd8adb9be71ae2043a6e  guix-build-1a332c78dbc7/output/riscv64-linux-gnu/SHA256SUMS.part
68fbe4bda4a3c9557a31d75c11193d736c9ff3e42fcb10ed77debb125b2be65b  guix-build-1a332c78dbc7/output/riscv64-linux-gnu/bitcoin-1a332c78dbc7-riscv64-linux-gnu-debug.tar.gz
92225f1f2587480b28caddd2c99b56096aab2788f280c0b6aca633902dda66a5  guix-build-1a332c78dbc7/output/riscv64-linux-gnu/bitcoin-1a332c78dbc7-riscv64-linux-gnu.tar.gz
8d6b05171b47855e57b6e979b38846be5fda9127aee6a7936760bbe3033c6db3  guix-build-1a332c78dbc7/output/x86_64-apple-darwin/SHA256SUMS.part
62bfcd5213cd49b3788ec915395457141d2bd25b48f0f22ada0f0edc10b041d7  guix-build-1a332c78dbc7/output/x86_64-apple-darwin/bitcoin-1a332c78dbc7-x86_64-apple-darwin-unsigned.dmg
8b9eccc66a1265a08fa8ea40f9fe7e4b9565c43ae98ecc556b6a62ec9aa0548c  guix-build-1a332c78dbc7/output/x86_64-apple-darwin/bitcoin-1a332c78dbc7-x86_64-apple-darwin-unsigned.tar.gz
b1a9673d34d9fbdf3d888ac56c61f30f0b3269ea373b345813d753e4d6ceb7ca  guix-build-1a332c78dbc7/output/x86_64-apple-darwin/bitcoin-1a332c78dbc7-x86_64-apple-darwin.tar.gz
98efad7c04c559f71af3adc4e8651bb39afc5bbdebe3cc364424b691ffd59c0d  guix-build-1a332c78dbc7/output/x86_64-linux-gnu/SHA256SUMS.part
5407f99b7345c33eeacb40596b7d78fe34b144194e4bad66b973fc13e6b90c58  guix-build-1a332c78dbc7/output/x86_64-linux-gnu/bitcoin-1a332c78dbc7-x86_64-linux-gnu-debug.tar.gz
8227eb1ace65aad6e6ed93027fb1c5a46c0f0aca673f3a2787a9579487ed2a3e  guix-build-1a332c78dbc7/output/x86_64-linux-gnu/bitcoin-1a332c78dbc7-x86_64-linux-gnu.tar.gz
847877454d16fa36c1ee024c59f015602f2c56f3370ac0506f0edce416a020ec  guix-build-1a332c78dbc7/output/x86_64-w64-mingw32/SHA256SUMS.part
b393ae8e1e54e49d496471b0859d88de05eb73dd6d024da98abfd8e9a4887c7b  guix-build-1a332c78dbc7/output/x86_64-w64-mingw32/bitcoin-1a332c78dbc7-win64-debug.zip
b0f0afef9eec68428b4f36cc70f8f7f69e1a516b94c53589f8923d9dd72c1288  guix-build-1a332c78dbc7/output/x86_64-w64-mingw32/bitcoin-1a332c78dbc7-win64-setup-unsigned.exe
a3929ca2e60059d5bacdfaa2bd06e3a899e91036df19a9fd3caafc866191bb87  guix-build-1a332c78dbc7/output/x86_64-w64-mingw32/bitcoin-1a332c78dbc7-win64-unsigned.tar.gz
2c943db51a2ac5d5ce3001df6de141fd88da25bc32dcf448926488336b902516  guix-build-1a332c78dbc7/output/x86_64-w64-mingw32/bitcoin-1a332c78dbc7-win64.zip
```

File: depends/packages/miniupnpc.mk [hunk 1]
```diff
@@ -3,17 +3,18 @@ $(package)_version=2.2.2
...
 $(package)_file_name=$(package)-$($(package)_version).tar.gz
 $(package)_sha256_hash=888fb0976ba61518276fe1eda988589c700a3f2a69d71089260d75562afd3687
-$(package)_patches=dont_leak_info.patch
+$(package)_patches=dont_leak_info.patch respect_mingw_cflags.patch
 
 define $(package)_set_vars
 $(package)_build_opts=CC="$($(package)_cc)"
 $(package)_build_opts_darwin=LIBTOOL="$($(package)_libtool)"
-$(package)_build_opts_mingw32=-f Makefile.mingw
+$(package)_build_opts_mingw32=-f Makefile.mingw CFLAGS="$($(package)_cflags) -D_WIN32_WINNT=0x0601"
```
Comment (luke-jr) [inline depends/packages/miniupnpc.mk:11]: Does this actually work?

`Makefile.mingw` defines `_WIN32_WINNT` in CPPFLAGS, which it puts on the commandline after CFLAGS, so I would expect it to override whatever we put in CFLAGS here...

Comment (fanquake) [reply depends/packages/miniupnpc.mk:11]: > Does this actually work?

Yes.

`make -C depends/ HOST=x86_64-w64-mingw32 -j9 miniupnpc V=1`.

master (a9049dd296a47319038d8c34bcc1ca46c89dee9c):
```bash
x86_64-w64-mingw32-gcc -Wall -W -Wstrict-prototypes -Os -DNDEBUG -D_WIN32_WINNT=0X501 -DMINIUPNP_STATICLIB -c -o upnpcommands.o upnpcommands.c
```

This PR (1a332c78dbc78a8fd941289c21e662b10bd3b914):
```bash
x86_64-w64-mingw32-gcc -pipe -std=c11 -O2 -D_WIN32_WINNT=0x0601 -DMINIUPNP_STATICLIB -c -o upnpcommands.o upnpcommands.c
```

**Review by jarolrod — APPROVED** [review]
ACK 1a332c78dbc78a8fd941289c21e662b10bd3b914

File: depends/packages/miniupnpc.mk (hunk 1 above)
Comment (luke-jr) [reply depends/packages/miniupnpc.mk:11]: Ah, apparently this part has changed in master miniupnpc. Maybe add a comment or assert-like check to ensure we don't bump it without fixing?

Comment (fanquake) [reply depends/packages/miniupnpc.mk:13]: I've added a comment for the next update.

**Review by jarolrod — COMMENTED** [review]
re-ACK 859644b

Only a comment has been added since my last review. Agree with pulling in this patch than the alternative of using the upstream master branch as source. Guix hashes match.

**Review by hebasto — APPROVED** [review]
ACK 859644b3c8550bb596fc8823747859e772dd493a, I've verified introduced changes in compiler flags, including the case with `DEBUG=1`.

It seems worth mentioning that 8e2d93ff0fee36baf7fe88662c83f5b4f40f0add also makes passing the correct optimization flags--`-O2` for release build and `-O1` for debug build--instead of `-Os`.

**Pull request merged**